x-pack/filebeat/cmd/agent.go

```diff
@@ -19,10 +19,17 @@ func filebeatCfg(rawIn *proto.UnitExpectedConfig, agentInfo *client.AgentInfo) (
 		return nil, fmt.Errorf("error creating input list from raw expected config: %w", err)
 	}
 
+	for iter := range modules {
+		if _, ok := modules[iter]["type"]; !ok {
+			modules[iter]["type"] = rawIn.Type
```

Comment:
Mostly because I'm paranoid. When I was testing, I ran into a lot of issues where the field was sometimes there and sometimes not, so I'd rather be safe.